scio-core/src/main/scala/com/spotify/scio/coders/instances/AvroCoders.scala

```diff
@@ -124,8 +125,15 @@ trait AvroCoders {
   def avroGenericRecordCoder: Coder[GenericRecord] =
     Coder.beam(new SlowGenericRecordCoder)
 
-  import org.apache.avro.specific.SpecificRecordBase
-  implicit def genAvro[T <: SpecificRecordBase]: Coder[T] =
+  implicit def avroSpecificRecordCoder[T <: SpecificRecord: ClassTag]: Coder[T] = {
+    val clazz = ScioUtil.classOf[T]
+    val schema = SpecificData.get().getSchema(clazz)
+    val useReflectApi = true // keep this for backward compatibility
+    Coder.beam(AvroCoder.of(clazz, schema, useReflectApi))
+  }
+
+  @deprecated("Use avroSpecificRecordCoder instead", since = "0.12.5")
+  def genAvro[T <: SpecificRecordBase]: Coder[T] =
```

Comment:
Drop this as done in #4590. This coder can't be called on a generic type with bounds.